src/rdf4cpp/util/Int128.hpp

```diff
@@ -0,0 +1,425 @@
+
+        assert(res.ec == std::errc{});
+
+        std::string_view const s{buf.data(), static_cast<std::string::size_type>(res.ptr - buf.data())};
+        return writer::write_str(s, writer);
+    }
+#endif
+
+    namespace detail {
+        enum struct OverflowMode {
+            Checked,
+            UndefinedBehavior,
+        };
+
+        template<std::size_t MinBits, std::size_t MaxBits, boost::multiprecision::cpp_integer_type SignType, typename Alloc>
+        using cpp_int_checked = boost::multiprecision::number<boost::multiprecision::cpp_int_backend<MinBits, MaxBits, SignType, boost::multiprecision::checked, Alloc>>;
+        template<std::size_t MinBits, std::size_t MaxBits, boost::multiprecision::cpp_integer_type SignType, typename Alloc>
+        using cpp_int_unchecked = boost::multiprecision::number<boost::multiprecision::cpp_int_backend<MinBits, MaxBits, SignType, boost::multiprecision::unchecked, Alloc>>;
+
+        template<typename T>
+        concept IntegralExt = std::integral<T> || std::same_as<T, __int128> || std::same_as<T, unsigned __int128>;
+        template<typename T>
+        concept BoostNumber = std::same_as<T, boost::multiprecision::number<typename T::backend_type, T::et>>;
+
+        template<OverflowMode m, typename T>
+        requires IntegralExt<T>
+        static constexpr bool add_checked(T const &a, T const &b, T &result) noexcept {
+            if constexpr (m == OverflowMode::Checked) {
+                return __builtin_add_overflow(a, b, &result);
+            } else {
+                result = a + b;
+                return false;
+            }
+        }
+        template<OverflowMode m, std::size_t MinBits, std::size_t MaxBits, boost::multiprecision::cpp_integer_type SignType, typename Alloc>
+        static constexpr bool add_checked(cpp_int_checked<MinBits, MaxBits, SignType, Alloc> const &a,
+                                          cpp_int_checked<MinBits, MaxBits, SignType, Alloc> const &b,
+                                          cpp_int_checked<MinBits, MaxBits, SignType, Alloc> &result) noexcept {
+            if (m == OverflowMode::Checked) {
+                try {
+                    result = a + b;
+                } catch (std::overflow_error const &) {
+                    return true;
+                } catch (std::range_error const &) {
+                    return true;
+                }
+                return false;
+            } else {
+                using Unc = cpp_int_unchecked<MinBits, MaxBits, SignType, Alloc>;
+                result = cpp_int_checked<MinBits, MaxBits, SignType, Alloc>{Unc{a} + Unc{b}};
+                return false;
+            }
+        }
+        template<OverflowMode m, std::size_t MinBits, std::size_t MaxBits, boost::multiprecision::cpp_integer_type SignType, typename Alloc>
+        static constexpr bool add_checked(cpp_int_unchecked<MinBits, MaxBits, SignType, Alloc> const &a,
```

Comment:
Are these functions with `cpp_int_unchecked` needed? I didn't find any usages with a quick search

Reply:
A quick test on the branch revealed that they are indeed all unused